src/Formula/Resolution.hs

```diff
@@ -194,8 +194,9 @@ removeNumberAtEmptyClause res@(Res (a,b,(Clause c,_)))
   | null c = Res (a,b,(Clause c,Nothing))
   | otherwise = res
 
-showResStep :: ResStep -> String
-showResStep = show . pretty . removeNumberAtEmptyClause
+showResSteps :: [ResStep] -> String
+showResSteps steps = "[" ++ intercalate ", " prettyMap ++ "]"
+  where prettyMap = map (show . pretty . removeNumberAtEmptyClause) steps
```

Comment:
I feel like this is reinventing the wheel. The showing of lists via `intercalate` etc. is something that `pretty :: [ ... ] -> Doc` is almost certainly already doing.

Reply:
`showResSteps = show . pretty . map removeNumberAtEmptyClause` does the same thing as your code?

Reply:
It does not do quite the same. `pretty` puts every list element into a new line. The output would look like this:
![image](https://github.com/fmidue/logic-tasks/assets/24428341/4d377e4b-6cb3-4063-ba51-7c487fdb4fd4)
(which I personally don't like in this case).
What's your preference?

Reply:
It's a pretty-printer, so it makes decisions about layout. The behavior that will be active here comes from: https://hackage.haskell.org/package/wl-pprint-text-1.2.0.2/docs/Text-PrettyPrint-Leijen-Text.html#v:list
> rendered horizontally if that fits the page ...

So, I'd say it's a problem with the `show`-defaults on the `Doc` type. It could be replaced by an explicit `renderPretty` with own settings (wider page), but even then, for long sequences the page width would be overrun and the pretty printer would switch to vertical layout.

On the other hand, the long format will never break a line. Is that really better layout?

I'd opt for trusting the pretty printer here, also since in Autotool it will not look quite as bad as
```
 <[(...
,(...
```
but more likely thus:
```
[(...
,(...
```

Reply:
It's using the pretty-printer now.